sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -98,6 +102,8 @@ typedef struct _nib_onl_entry {
      * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0.
      */
     uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+    evtimer_msg_event_t reach_timeout;  /**< Event for @ref GNRC_IPV6_NIB_REACH_TIMEOUT */
+    evtimer_msg_event_t delay_timeout;  /**< Event for @ref GNRC_IPV6_NIB_DELAY_TIMEOUT */
```

Comment:
>Also regarding non-routing 6LN: they most likely only need 1 on-link entry

And for routing 6LNs you probably need as much as you can get.

Could you please revisit the need for all `14` event types? I understand there is no configuration where all `14` are used on a node, but it still has a high impact. Having a quick look at [1], is it e.g. possible to merge `ROUTE_TIMEOUT` and `PREFIX_TIMEOUT`? Are there any other event types that could be merged and multiplexed when an event is handled?

[1] https://miri64.github.io/riot-ndp-model/#api